athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/serde/v5/ConstraintsSerDeV5.java

```diff
@@ -150,7 +159,13 @@ public Constraints doDeserialize(JsonParser jparser, DeserializationContext ctxt
                 queryPassthroughArguments.put(jparser.getCurrentName(), jparser.getValueAsString());
             }
 
-            Constraints constraints = new Constraints(summaryMap.build(), federationExpression.build(), orderByClauseBuilder.build(), limit, queryPassthroughArguments);
+            QueryPlan queryPlan = null;
+            assertFieldName(jparser, QUERY_PLAN);
+            if (jparser.nextToken() != JsonToken.VALUE_NULL) {
```

Comment:
You need to create a ConstraintsSerDeV6 with these changes. Otherwise we'll run into issues when we serialize with a V5 client and the connector is using V6